hibernate-core/src/main/java/org/hibernate/collection/internal/StandardSortedMapSemantics.java

```diff
@@ -49,7 +49,9 @@ public CollectionClassification getCollectionClassification() {
 			Object key,
 			CollectionPersister collectionDescriptor,
 			SharedSessionContractImplementor session) {
-		return new PersistentSortedMap<>( session );
+		final PersistentSortedMap<K,V> result = new PersistentSortedMap<>( session );
+		result.setComparator( (Comparator<K>) collectionDescriptor.getSortingComparator() );
```

Comment:
It seems to me that the setter `setComparator` is unnecessary and should be removed. IMO the comparator should be `final` in `PersistentSortedMap` as well as `PersistentSortedSet` and be initialized through the constructor.

Reply:
We can certainly look at passing it via the ctor so long as are willing to remove the ctor that takes the `SessionImplementor` argument that is deprecated.  The setter however needs to remain to support SOAP libraries, as indicated in the no-arg ctor.